perun-core/src/test/java/cz/metacentrum/perun/core/entry/AttributesManagerEntryIntegrationTest.java

```diff
@@ -5889,6 +5892,56 @@ public void deleteAttributeForceWhenRelationExists() throws Exception {
 
 	}
 
+	@Test
+	public void deleteAttributeCorrectlyRemovesModules() throws Exception {
+		System.out.println(CLASS_NAME + "deleteAttributeCorrectlyRemovesModules");
+		String namespaceAAA = "AAA";
+		String namespaceBBB = "BBB";
+
+		// need this to get AttributesManagerImpl
+		PerunBlImpl perunBlImpl = ((PerunBlImpl) perun);
+
+
+		//create attribute gid in namespace aaa
+		AttributeDefinition g_gid_AAA_def = new AttributeDefinition();
+		g_gid_AAA_def.setNamespace(AttributesManager.NS_GROUP_ATTR_DEF);
+		g_gid_AAA_def.setDescription("gid in namespace AAA");
+		g_gid_AAA_def.setFriendlyName("unixGIDtest-namespace:" + namespaceAAA);
+		g_gid_AAA_def.setType(Integer.class.getName());
+		g_gid_AAA_def = perun.getAttributesManagerBl().createAttribute(sess, g_gid_AAA_def);
+
+		//create attribute gid in namespace bbb
+		AttributeDefinition g_gid_BBB_def = new AttributeDefinition();
+		g_gid_BBB_def.setNamespace(AttributesManager.NS_GROUP_ATTR_DEF);
+		g_gid_BBB_def.setDescription("gid in namespace BBB");
+		g_gid_BBB_def.setFriendlyName("unixGIDtest-namespace:" + namespaceBBB);
+		g_gid_BBB_def.setType(Integer.class.getName());
+		g_gid_BBB_def = perun.getAttributesManagerBl().createAttribute(sess, g_gid_BBB_def);
+		AttributesModuleImplApi bbbModule = (AttributesModuleImplApi) perunBlImpl.getAttributesManagerImpl().getAttributesModule(sess, g_gid_BBB_def);
+
+		//create common attribute gid
+		AttributeDefinition g_gid_common_def = new AttributeDefinition();
+		g_gid_common_def.setNamespace(AttributesManager.NS_GROUP_ATTR_DEF);
+		g_gid_common_def.setDescription("common gid");
+		g_gid_common_def.setFriendlyName("unixGIDtest-namespace");
+		g_gid_common_def.setType(Integer.class.getName());
+		g_gid_common_def = perun.getAttributesManagerBl().createAttribute(sess, g_gid_common_def);
+		AttributesModuleImplApi commonModule = (AttributesModuleImplApi) perunBlImpl.getAttributesManagerImpl().getAttributesModule(sess, g_gid_common_def);
+
+		attributesManager.deleteAttribute(sess, g_gid_AAA_def);
+		// should only delete namespace-specific, since BBB remains
+		assertNull(perunBlImpl.getAttributesManagerImpl().getAttributesModule(sess, g_gid_AAA_def));
+		assertEquals(perunBlImpl.getAttributesManagerImpl().getAttributesModule(sess, g_gid_BBB_def), bbbModule);
```

Comment:
All the attribute modules (AAA, BBB, common) are null in these checks, so the test is not covering what it's supposed to, it passes on null equals null. Please look at it.